src/Controls/src/Core/HandlerImpl/Application/Application.Impl.cs

```diff
@@ -30,7 +31,10 @@ IWindow IApplication.CreateWindow(IActivationState? activationState)
 			if (activationState?.State?.TryGetValue(MauiWindowIdKey, out var requestedWindowId) ?? false)
 			{
 				if (requestedWindowId != null && _requestedWindows.TryGetValue(requestedWindowId, out var w))
```

Comment:
I'm testing this on Android and the `TryGetValue` here seems like it's always evaluating to false.

Reply:
I added a test for this issue, and changed it to a `Dictionary<string, WeakReference<Window>>` which won't cause this behavior.

I think you were getting a new `string` instance and `ConditionalWeakTable` doesn't work in that case.

https://github.com/dotnet/maui/blob/2e7c3ff89e3a0e9a6dfeaf25149508c3b0d1b533/src/Controls/tests/Core.UnitTests/WindowsTests.cs#L664-L670